src/jit/codegenarm64.cpp

```diff
@@ -4578,49 +4578,139 @@ void CodeGen::genSIMDIntrinsicGetItem(GenTreeSIMD* simdNode)
     // - the source of SIMD type (op1)
     // - the index of the value to be returned.
     genConsumeOperands(simdNode);
-    regNumber srcReg = op1->gtRegNum;
 
-    // TODO-ARM64-CQ Optimize SIMDIntrinsicGetItem
-    // Optimize the case of op1 is in memory and trying to access ith element.
-    assert(op1->isUsedFromReg());
+    emitAttr baseTypeSize  = emitTypeSize(baseType);
+    unsigned baseTypeScale = genLog2(EA_SIZE_IN_BYTES(baseTypeSize));
 
-    emitAttr baseTypeSize = emitTypeSize(baseType);
-
-    if (op2->IsCnsIntOrI())
+    if (op1->isContained())
     {
-        assert(op2->isContained());
+        // Optimize the case of op1 is in memory and trying to access ith element.
+        assert(!op1->isUsedFromReg());
+
+        if (op1->OperIsLocal())
+        {
+            unsigned varNum = op1->gtLclVarCommon.gtLclNum;
+
+            if (op2->IsCnsIntOrI())
+            {
+                assert(op2->isContained());
 
-        ssize_t index = op2->gtIntCon.gtIconVal;
+                ssize_t index = op2->gtIntCon.gtIconVal;
 
-        if (getEmitter()->isValidVectorIndex(emitTypeSize(simdType), baseTypeSize, index))
+                if (getEmitter()->isValidVectorIndex(emitTypeSize(simdType), baseTypeSize, index))
```

Comment:
Shouldn't there be an `else` to this `if`? That is, it seems to me that this condition should either be evaluated when the decision is made to contain the constant, and/or code should be generated here to directly throw. (Or, perhaps it's the case that the check, which should have been generated to execute prior to this, will already have caused the throw?)

Reply:
> it's the case that the check, which should have been generated to execute prior to this, will already have caused the throw?

Yes

Reply:
Right, so perhaps a comment to that effect would be good.

Reply:
Comment is a few lines below in the block